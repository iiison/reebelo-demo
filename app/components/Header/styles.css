.visaLogo {
  width: 50px;
  height: auto;
  margin: 10px 5px 5px 5px;
}

.header {
  border-bottom: 1px solid #999A99;
  position: relative;
}

.back{
  position: absolute;
  height: 100%;
  width: 20px;
  left: 12px;

  &:before,
  &:after {
    content: '';
    position: absolute;
    background: #999A99;
    height: 3px;
    width: 15px;
    left: 0;
  }

  &:before {
    transform: rotate(-45deg);
    top: 12px;
  }

  &:after {
    transform: rotate(45deg);
    top: 21px;
  }
}
