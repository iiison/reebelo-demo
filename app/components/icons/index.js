export ShowIcon from './password-show.svg';
export HideIcon from './password-hide.svg';
export VisaLogo from './visa_logo.svg';
export VisaSuccessLogo from './visa-success-logo.png';
