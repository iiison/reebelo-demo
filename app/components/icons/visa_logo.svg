<svg id="Visa_Logo" style="background-color: #fafafa;" data-name="Visa Logo" xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 216 69.76"><defs><style>.cls-1{fill:url(#Visa_Blue_Gradient);}</style><linearGradient id="Visa_Blue_Gradient" x1="-0.82" y1="66.23" x2="215.8" y2="4.12" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#1a1e5a"/><stop offset="1" stop-color="#122d98"/></linearGradient></defs><title>visa_vbm_blugrad</title><path class="cls-1" d="M82,1.23,53.73,68.7H35.28L21.37,14.86c-.85-3.32-1.58-4.53-4.15-5.93C13,6.65,6.09,4.52,0,3.19l.41-2h29.7a8.13,8.13,0,0,1,8,6.88l7.35,39L63.68,1.23ZM154.3,46.67c.08-17.8-24.62-18.78-24.45-26.74.05-2.42,2.36-5,7.4-5.65a32.88,32.88,0,0,1,17.21,3L157.52,3A47,47,0,0,0,141.2,0C124,0,111.8,9.17,111.7,22.3,111.59,32,120.37,37.44,127,40.66c6.8,3.31,9.09,5.43,9.06,8.39,0,4.53-5.43,6.52-10.45,6.6-8.77.14-13.86-2.37-17.92-4.26L104.5,66.17c4.08,1.87,11.61,3.51,19.41,3.59,18.34,0,30.34-9.06,30.39-23.09m45.56,22H216L201.91,1.23H187a7.94,7.94,0,0,0-7.43,4.95L153.39,68.7h18.33l3.64-10.08h22.39ZM180.38,44.8l9.19-25.34,5.29,25.34ZM107,1.23,92.52,68.7H75.06L89.5,1.23Z"/></svg>