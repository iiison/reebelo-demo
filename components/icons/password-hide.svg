<svg id="Layer_1" style="background-color: #fafafa;" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><title>password-hide</title><path d="M13.93,11.48l-2.45,2.45A1.81,1.81,0,0,0,12,14a2,2,0,0,0,2-2A1.81,1.81,0,0,0,13.93,11.48Zm-1.41-1.41A1.81,1.81,0,0,0,12,10a2,2,0,0,0-2,2,1.81,1.81,0,0,0,.07.52Zm5.31-2.49L15.45,10A4,4,0,0,1,10,15.45L7.91,17.5a9.07,9.07,0,0,0,4.09,1c3.64,0,7.11-2.3,9.82-6.5A16.78,16.78,0,0,0,17.83,7.58ZM16.09,6.5A9.07,9.07,0,0,0,12,5.5C8.36,5.5,4.89,7.8,2.18,12a16.78,16.78,0,0,0,4,4.42L8.55,14A4,4,0,0,1,14,8.55ZM6.43,19,1.71,23.71A1,1,0,0,1,.29,22.29l4.45-4.44A19.46,19.46,0,0,1,.15,12.52a1,1,0,0,1,0-1c3.12-5.15,7.33-8,11.85-8A11.19,11.19,0,0,1,17.57,5L22.29.29a1,1,0,1,1,1.42,1.42L19.26,6.15a19.63,19.63,0,0,1,4.59,5.33,1,1,0,0,1,0,1c-3.12,5.15-7.33,8-11.85,8A11.19,11.19,0,0,1,6.43,19Z"/></svg>