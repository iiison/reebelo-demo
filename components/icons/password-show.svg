<svg id="Layer_1" style="background-color: #fafafa;" data-name="Layer 1" xmlns="http://www.w3.org/2000/svg" viewBox="0 0 24 24"><title>password-show</title><path d="M12,10a2,2,0,1,0,2,2A2,2,0,0,0,12,10Zm0,6a4,4,0,1,1,4-4A4,4,0,0,1,12,16ZM2.18,12c2.71,4.2,6.18,6.5,9.82,6.5s7.11-2.3,9.82-6.5C19.11,7.8,15.64,5.5,12,5.5S4.89,7.8,2.18,12ZM12,20.5c-4.52,0-8.73-2.83-11.85-8a1,1,0,0,1,0-1c3.12-5.15,7.33-8,11.85-8s8.73,2.83,11.85,8a1,1,0,0,1,0,1C20.73,17.67,16.52,20.5,12,20.5Z"/></svg>