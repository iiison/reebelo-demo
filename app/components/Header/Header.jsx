import React from 'react';

import styles from './styles.css'

const Header = () => { 
  return ( 
    <header className={`col-12 t-left padded-l ${styles.header}`}>
      <h1 className='header-color'>Demo</h1>
    </header>
  )
}

export default Header
