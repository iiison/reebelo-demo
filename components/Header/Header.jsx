import React from 'react';
import Link from 'next/link'

import styles from './styles.module.css'

const Header = () => { 
  return ( 
    <header className={`col-12 t-left padded-l ${styles.header}`}>
      <Link href='/' passHref>
        <h1 className='header-color'>Demo</h1>
      </Link>
    </header>
  )
}

export default Header
