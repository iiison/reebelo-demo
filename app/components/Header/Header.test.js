import React from 'react'
import { shallow } from 'enzyme'
import renderer from 'react-test-renderer'

import Header from './Header'
import { VisaLogo } from '$COMPONENTS/icons/index';

describe('>> COMPONENT -- Tests Header', () => {
    it('Header Snapshot Test', () => {
        const component = renderer
            .create(<Header />)
            .toJSON()

        expect(component).toMatchSnapshot();
    })

    it('renders visa logo', () => {
        const header = shallow(<Header />);

        expect(header.find("img").prop("src")).toEqual(VisaLogo);

    });

    it('renders hr component', () => {
        const header = shallow(<Header />);

        expect(header.containsMatchingElement(<hr />)).toEqual(true)
    })
})

