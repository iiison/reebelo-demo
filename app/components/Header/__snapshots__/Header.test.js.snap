// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`>> COMPONENT -- Tests Header Header Snapshot Test 1`] = `
<div
  className="col-12 t-center "
>
  <img
    className="visaLogo"
    src={Object {}}
  />
  <hr
    className="divider"
  />
</div>
`;
